From 2c7c331b747cb567e345d2246213b5a3b755a42b Mon Sep 17 00:00:00 2001
From: Kazuya Mizuguchi <kazuya.mizuguchi.ks@renesas.com>
Date: Fri, 18 May 2018 18:13:48 +0900
Subject: [PATCH 194/909] ravb: Fix register define for Gen3

commit 621eac5bf5e19da6c80643f191ab0d3aec402884 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

This patch fixes registers defined for Gen3.

Signed-off-by: Kazuya Mizuguchi <kazuya.mizuguchi.ks@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/net/ethernet/renesas/ravb.h | 51 ++++++++++++++++++++++++++++-
 1 file changed, 50 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/renesas/ravb.h b/drivers/net/ethernet/renesas/ravb.h
index 81e9141fa5c4..1bfa5e17cd9e 100644
--- a/drivers/net/ethernet/renesas/ravb.h
+++ b/drivers/net/ethernet/renesas/ravb.h
@@ -179,6 +179,7 @@ enum ravb_reg {
 	TFA0	= 0x0310,
 	TFA1	= 0x0314,
 	TFA2	= 0x0318,
+	VRR	= 0x031C,	/* R-Car Gen3 only */
 	CIVR0	= 0x0320,
 	CIVR1	= 0x0324,
 	CDVR0	= 0x0328,
@@ -250,6 +251,8 @@ enum ravb_reg {
 	EID	= 0x045C,	/* R-Car Gen3 only */
 	RIE0	= 0x0460,	/* R-Car Gen3 only */
 	RID0	= 0x0464,	/* R-Car Gen3 only */
+	RIE1	= 0x0468,	/* R-Car Gen3 only */
+	RID1	= 0x046C,	/* R-Car Gen3 only */
 	RIE2	= 0x0470,	/* R-Car Gen3 only */
 	RID2	= 0x0474,	/* R-Car Gen3 only */
 	TIE	= 0x0478,	/* R-Car Gen3 only */
@@ -269,7 +272,7 @@ enum ravb_reg {
 	MPR	= 0x0558,
 	PFTCR	= 0x055c,
 	PFRCR	= 0x0560,
-	PFTTLR	= 0x0564,	/* R-Car Gen3 only */
+	TPAUSER	= 0x0564,	/* R-Car Gen3 only */
 	PFTTCR	= 0x0568,	/* R-Car Gen3 only */
 	GECMR	= 0x05b0,
 	MAHR	= 0x05c0,
@@ -812,6 +815,52 @@ enum RID0_BIT {
 	RID0_FRD17	= 0x00020000,
 };
 
+/* RIE1 (R-Car Gen3 only) */
+enum RIE1_BIT {
+	RIE1_RWS0	= 0x00000001,
+	RIE1_RWS1	= 0x00000002,
+	RIE1_RWS2	= 0x00000004,
+	RIE1_RWS3	= 0x00000008,
+	RIE1_RWS4	= 0x00000010,
+	RIE1_RWS5	= 0x00000020,
+	RIE1_RWS6	= 0x00000040,
+	RIE1_RWS7	= 0x00000080,
+	RIE1_RWS8	= 0x00000100,
+	RIE1_RWS9	= 0x00000200,
+	RIE1_RWS10	= 0x00000400,
+	RIE1_RWS11	= 0x00000800,
+	RIE1_RWS12	= 0x00001000,
+	RIE1_RWS13	= 0x00002000,
+	RIE1_RWS14	= 0x00004000,
+	RIE1_RWS15	= 0x00008000,
+	RIE1_RWS16	= 0x00010000,
+	RIE1_RWS17	= 0x00020000,
+	RIE1_RFWS	= 0x80000000,
+};
+
+/* RID1 (R-Car Gen3 only) */
+enum RID1_BIT {
+	RID1_RWD0	= 0x00000001,
+	RID1_RWD1	= 0x00000002,
+	RID1_RWD2	= 0x00000004,
+	RID1_RWD3	= 0x00000008,
+	RID1_RWD4	= 0x00000010,
+	RID1_RWD5	= 0x00000020,
+	RID1_RWD6	= 0x00000040,
+	RID1_RWD7	= 0x00000080,
+	RID1_RWD8	= 0x00000100,
+	RID1_RWD9	= 0x00000200,
+	RID1_RWD10	= 0x00000400,
+	RID1_RWD11	= 0x00000800,
+	RID1_RWD12	= 0x00001000,
+	RID1_RWD13	= 0x00002000,
+	RID1_RWD14	= 0x00004000,
+	RID1_RWD15	= 0x00008000,
+	RID1_RWD16	= 0x00010000,
+	RID1_RWD17	= 0x00020000,
+	RID1_RFWD	= 0x80000000,
+};
+
 /* RIE2 (R-Car Gen3 only) */
 enum RIE2_BIT {
 	RIE2_QFS0	= 0x00000001,
-- 
2.17.1

